libbeat/docs/outputconfig.asciidoc

```diff
@@ -527,9 +565,57 @@ The password for connecting to Kafka.
 
 ===== topic
 
-The Kafka topic used for produced events. The setting can be a format string
+The Kafka topic used for produced events. The setting can be a *Format String*
 using any event field. To set the topic from document type use `%{[type]}`.
 
+===== topics
+
+Array of topic selector rules supporting conditionals, *Format String*
+based field access and name mappings. The first rule matching will be used to
+set the `topic` for the event to be published. If `topics` is missing or no
+rule matches, the `topic` field will be used.
+
+Rule settings:
+
+*`topic`*: The topic *Format String* to use. If the fields used are missing, the
+ rule fails.
+
+*`mapping`*: Dictionary mapping index names to new names
+
+*`default`*: Default string value if `mapping` does not find a match.
+
+*`when`*: Condition which must succeed in order to execute the current rule.
+
+===== key
+
+The Kafka event key. The event key must be unique and can be extracted from the
+event using a *Format String*.
+
+===== partition
+
+Kafka output broker event partitioning strategy. Must be one of `random`,
+`round_robin`, or `hash`. By default the `hash` partitioner is used.
+
+*`random.group_events`*: Select a new partition by random every `group_events`
+ events being published. The default value is 1 meaning after each event a new
+ parition is picked randomly.
+
+*`round_robin.group_events`*: Select the next partition every `group_events`
+ events being published. The default value is 1 meaning after each event the next partition will be selected.
+
+*`hash.hash`*: List of fields used to compute the partitioning hash value from.
+ If no field is configured, the events `key` value will be used.
+
+*`hash.random`*: Randomly distribute events if no hash or key value can be computed.
+
+All partitioners will try to publish events to all partitions by default. If a
+partitions leader becomes unreachable for the beat, the output might block. All
+partitioners support setting `reachable_only` to overwrite this
+behavior. If `reachable_only` is set to `true`, events will be published to
+available partitions only. Note: publishing to the subset of available partitions
```

Comment:
Make this a proper note by starting it on a separate line and beginning with "NOTE: ". Also, saying "only" makes the sentence grammatically ambiguous in English. I would say something like:

NOTE: Publishing to a subset of available partitions potentially increases resource usage because events may become unevenly distributed.